tests/flows/test_index.py

```diff
@@ -321,6 +324,123 @@ async def test_run_partial_updates_of_concepts_for_document_passages(
+        Bucket=mock_bucket, Key=key, Body=body, ContentType="application/json"
+    )
+
+    # Setup Vespa
+
+    subprocess.run(
+        [
+            "vespa",
+            "feed",
+            "tests/local_vespa/test_documents/family_document_UNFCCC.party.309.0.json",
+        ],
+        capture_output=True,
+        text=True,
+        check=True,
+        timeout=600,  # Seconds
+    )
+
+    subprocess.run(
+        [
+            "vespa",
+            "feed",
+            "tests/local_vespa/test_documents/document_passage_UNFCCC.party.309.0.json",
+        ],
+        capture_output=True,
+        text=True,
+        check=True,
+        timeout=600,  # Seconds
+    )
+
+    # Make sure the expected document passages are in Vespa
+    async with local_vespa_search_adapter.client.asyncio() as vespa_connection_pool:
+        # Confirm that the example concepts are not in the document passages
+        initial_passages = await get_document_passages_from_vespa(
+            document_import_id=document_import_id,
+            text_blocks_ids=None,
+            vespa_connection_pool=vespa_connection_pool,
+        )
+        assert len(initial_passages) == 50_000
+
+    # Confirm that we can add the example concepts to the document passages
+    #
+    # The model names should be different. It's to do with bad fixture setup by us.
+    test_counts = Counter(
+        {
+            ConceptModel(
+                wikibase_id=WikibaseID("Q10015"),
+                model_name='KeywordClassifier("professional services sector")',
+            ): 2,
+            ConceptModel(
+                wikibase_id=WikibaseID("Q10014"),
+                model_name='KeywordClassifier("professional services sector")',
+            ): 1,
+        }
+    )
+
+    (
+        merge_serialise_concepts_cb,
+        vespa_response_handler_cb,
+        concepts_counts_updater_cb,
+    ) = op_to_fn(Operation.INDEX)
+
+    assert (
+        test_counts
+        == await run_partial_updates_of_concepts_for_document_passages.fn(
+            document_importer=(document_import_id, document_object_uri),
+            vespa_search_adapter=local_vespa_search_adapter,
+            cache_bucket=mock_bucket,
+            concepts_counts_prefix=CONCEPTS_COUNTS_PREFIX_DEFAULT,
+            merge_serialise_concepts_cb=merge_serialise_concepts_cb,
+            vespa_response_handler_cb=vespa_response_handler_cb,
+            concepts_counts_updater_cb=concepts_counts_updater_cb,
+        )
+    )
+    async with local_vespa_search_adapter.client.asyncio() as vespa_connection_pool:
+        updated_passages = await get_document_passages_from_vespa(
+            document_import_id=document_import_id,
+            text_blocks_ids=None,
+            vespa_connection_pool=vespa_connection_pool,
+        )
+    updated_concepts = [
+        concept
+        for _, passage in updated_passages
+        if passage.concepts
+        for concept in passage.concepts
+    ]
```

Comment:
thought: We could add in a comparison of the initial and updated concepts

Reply:
✓